File: solutions/python/dnd-character/1/dnd_character.py
class Character:
    def __init__(self):
        self.strength = Character.dados()
        self.dexterity = Character.dados()
        self.constitution = Character.dados()
        self.intelligence = Character.dados()
        self.wisdom = Character.dados()
        self.charisma = Character.dados()
        self.hitpoints = 10 + modifier(self.constitution)
    def dados():
        import random
        dados = []
        for _ in range(4):
            dados.append(random.randint(1,2))
        del dados[dados.index(min(dados))]
        return sum(dados)
    def ability(self):
        return self.constitution

def modifier(n):
    return (n - 10) // 2

File: solutions/javascript/dnd-character/1/dnd-character.js
//
// This is only a SKELETON file for the 'D&D Character' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const abilityModifier = (stat) => {
  if (stat < 3) throw new Error("Ability scores must be at least 3");
  if (stat > 18) throw new Error("Ability scores can be at most 18");
  return Math.floor((stat - 10) / 2);
};

export class Character {
  constructor() {
    this.str = Character.rollAbility();
    this.dex = Character.rollAbility();
    this.con = Character.rollAbility();
    this.int = Character.rollAbility();
    this.wis = Character.rollAbility();
    this.cha = Character.rollAbility();
    this.hp = 10 + abilityModifier(this.con);
  }

  static rollAbility() {
    const rolls = [...Array(4)].map(() => Math.floor(Math.random() * 6) + 1);
    rolls.sort((a, b) => a - b);
    return rolls.slice(1).reduce((a, b) => a + b);
  }

  get strength() {
    return this.str;
  }

  get dexterity() {
    return this.dex;
  }

  get constitution() {
    return this.con;
  }

  get intelligence() {
    return this.int;
  }

  get wisdom() {
    return this.wis;
  }

  get charisma() {
    return this.cha;
  }

  get hitpoints() {
    return this.hp;
  }
}
